presto-main/src/main/java/com/facebook/presto/server/TaskInfoResource.java

```diff
@@ -69,4 +118,34 @@ public TaskInfo getTaskInfo(@PathParam("taskId") TaskId taskId)
             throw new NotFoundException(e);
         }
     }
+
+    private void proxyTaskInfo(HttpServletRequest servletRequest, AsyncResponse asyncResponse, String xForwardedProto, UriInfo uriInfo)
+    {
+        try {
+            checkState(proxyHelper.isPresent());
+            Optional<InternalNode> resourceManager = internalNodeManager.getResourceManagers().stream()
+                    .findFirst();
+            if (resourceManager.isPresent()) {
+                InternalNode resourceManagerNode = resourceManager.get();
+                URI uri = uriInfo.getRequestUriBuilder()
```

Comment:
It is a bit strange to use the uriInfo of the coordinator to construct the uri of RM. Can't we directly use the UriBuilder

Reply:
Rewrote to use UriBuilder